City-of-Helsinki/drupal-helfi-etusivu
# UHF-8216: News RSS feed filters

# [UHF-8216](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8216)

News RSS feed for news archive using filters

## What was done


* Added link to RSS feed based on selected filters
* Replaced custom datatype for main image with a custom processor
* Reworked aggregations into `multi_term` queries
* Unified query logic between RSS feed and react app

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-8216_News_RSS_feed_filters`
  * `make fresh`
* Run `make drush-cr`

Set up and rebuild Elastic index:
- Run `make up`. Check which port your elastic is running in with `make ps`
- Add this ☝️ to your .env file like so: `ELASTIC_PROXY_URL=http://localhost:[ELASTIC_PORT]`
- Run `make up` again to apply

Create / rebuild index:
- Run `drush sapi-rt; drush sapi-c; drush sapi-i` 

## How to test


Navigate to https://helfi-etusivu.docker.so/fi/uutiset to view the app and test it.

React App:
* [x] Open news archive and add / change some filters
* [x] RSS Feed subscribe link updates with selected / changed filters
* [x] RSS Feed url updates on `<head>` section while changing the filters
* [x] RSS Feed subscribe link opens correct path for current language

RSS feed:
Easiest way to test this is to use Postman or similar tool that lets you see the XML from RSS feed. Both the RSS feed and archive app should yield the same results with filters applied. Test this by making some searches with the archive and then copypasting the query string to the RSS url https://helfi-etusivu.docker.so/fi/uutiset/rss. For example, `https://helfi-etusivu.docker.so/fi/uutiset/rss?topic[0]=397&groups[0]=2` and `https://helfi-etusivu.docker.so/fi/uutiset?topic[0]=397&groups[0]=2` should give you the same set of results.


## Designers review



## Other PRs


* [UHF-8216: Styles for News RSS feed link](https://github.com/City-of-Helsinki/drupal-hdbt/pull/645)


[UHF-8216]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
